<!DOCTYPE html>
<html lang="ko" xmlns:th="http://www.thymeleaf.org">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<script>
    // setTimeout() : 특정 시간 이후에 실행되는 함수를 받음
    setTimeout(() => console.log("실행"), 1000);
    console.log("code 6");

    // promise : 특정 함수가 실행된 이후에 실행될 함수를 처리
    new Promise(resolve => {
        setTimeout(() => {
            resolve(100)
        }, 3000);
    }).then(res => console.log(res));
    console.log("code 7");
    console.log("code 8");

    // promise에서 오류가 났을 때는 catch 실행
    // 무조건 거절이라 "에러" 뜸
    new Promise((resolve, reject) => {
        setTimeout(() => {
            reject(100)
        }, 3000);
    }).then(res => console.log("성공"))
        .catch((err) => console.log("에러"));

</script>
</body>
</html>